data/data/azure/dns/dns.tf

```diff
@@ -3,29 +3,26 @@ locals {
   api_external_name = "api.${replace(var.cluster_domain, ".${var.base_domain}", "")}"
 }
 
-resource "azureprivatedns_zone" "private" {
-  name                = var.cluster_domain
-  resource_group_name = var.resource_group_name
-}
-
 resource "azureprivatedns_zone_virtual_network_link" "network" {
+  count = var.azure_supports_private_dns ? 1 : 0
+
   name                  = "${var.cluster_id}-network-link"
   resource_group_name   = var.resource_group_name
-  private_dns_zone_name = azureprivatedns_zone.private.name
+  private_dns_zone_name = var.azure_cluster_dns_zone_name
   virtual_network_id    = var.virtual_network_id
 }
 
-resource "azureprivatedns_a_record" "apiint_internal" {
+resource "azurerm_dns_a_record" "apiint_internal" {
```

Comment:
This nukes private DNS. A method to use legacy DNS and new private DNS will need to happen for all private DNS records in this file.

Reply:
I believe this version of the Azurerm resource dns_a_record will create an a record in either type dns zone. Please see the reference: https://github.com/openshift/installer/blob/release-4.2/data/data/azure/dns/dns.tf. I can add additional resources based off the supports_private_dns. I am waiting on #2745 to upgrade azurerm module to the latest version.

Reply:
What this is doing, is removing the ability to create the newer private DNS records and reverting to the previous Azure DNS that will break installing into existing vnets. You'll need to have both types to support your environment. You can use the "azure_supports_private_dns" to distinguish which type of records to create.